### PR TITLE
Upgrade normalize for number() > int32

### PR DESCRIPTION
**Pull request by DrownFish19**
### PR Category

Operator Mechanism

### PR Types

Improvements

### Description

support normalize() for tensor's number > int32.

#### 精度和速度测试代码
```python
import torch
import paddle
import numpy as np
import time
 
p = 2.0
axis = -1
epsilon = 1e-12
input = np.random.randn(1024, 256, 64, 100, 2)
print(input.shape)
np.save("input.npy", input)

start_time = 0
end_time = 0

input = np.load("input.npy")
x_paddle = paddle.to_tensor(input)
print(x_paddle.numel())

# 仅取中间结果作为性能标准
for i in range(100):
    if i == 10:
        start_time = time.time()
    elif i == 90:
        end_time = time.time()
    paddle.nn.functional.normalize(x_paddle, float(p), axis)
    if i==99:
        out_paddle = paddle.nn.functional.normalize(x_paddle, float(p), axis)
np.save("out_paddle.npy", out_paddle.numpy())
print(f"paddle time cost {(end_time-start_time)/ 90}")

input = np.load("input.npy")
x_torch = torch.from_numpy(input).cuda(1)

# 仅取中间结果作为性能标准
for i in range(100):
    if i == 10:
        start_time = time.time()
    elif i == 90:
        end_time = time.time()
    torch.nn.functional.normalize(x_torch, float(p), axis)
    if i==99:
        out_torch = torch.nn.functional.normalize(x_torch, float(p), axis)
np.save("out_torch.npy", out_torch.cpu().numpy())
print(f"pytorch time cost {(end_time-start_time)/ 90}")

diff = np.load("out_paddle.npy") - np.load("out_torch.npy")
print(diff.max())
print(diff.min())
print(diff.mean())
print(diff.sum())
```

#### 精度和速度测试结果 (A100)
```text
Tensor(shape=[], dtype=int64, place=Place(gpu:0), stop_gradient=True,
       3355443200)
paddle time cost 3.959337870279948e-05
pytorch time cost 0.06611543231540257
4.440892098500626e-16
-4.440892098500626e-16
7.802966542400176e-22
2.6182411024524182e-12
```

#### 精度和速度测试结果 (H20)
```
Tensor(shape=[], dtype=int64, place=Place(gpu:0), stop_gradient=True,
       3355443200)
paddle time cost 2.6347902086046006e-05
pytorch time cost 0.17290243042839898
4.440892098500626e-16
-4.440892098500626e-16
-1.365545043940858e-21
-4.582008831985053e-12
```

Pcard-67164

### CODE REVIEW

**Review by wanghuancoder — APPROVED** [review]
LGTM
